object-construction-checker/tests/autovalue/Validation.java

```diff
@@ -0,0 +1,44 @@
+import com.google.auto.value.AutoValue;
+import org.checkerframework.checker.objectconstruction.qual.*;
+import org.checkerframework.checker.nullness.qual.*;
+
+
+@AutoValue
+abstract class Validation {
+
+  public abstract String name();
+
+  static Builder builder() {
+    return new AutoValue_Validation.Builder();
+  }
+
+  @AutoValue.Builder
+  abstract static class Builder {
+
+    abstract Builder setName(String name);
+
+    abstract Validation autoBuild();
+
+    public Validation build(@CalledMethods("setName") Builder this) {
```

Comment:
I don't understand why this `@CalledMethods` annotation shouldn't be inferred.

Reply:
Now we infer a `@CalledMethods` annotation for `autoBuild()`, since that's the method whose implementation gets generated by AutoValue.  We can't infer anything for this other public `build()` method since the developer wrote the code and we don't know what it does without analyzing the code.  In principle, it could set all the properties itself, in which case `@CalledMethodsTop` is a fine type for its receiver.  Make sense?

Reply:
Yes, thanks for explaining.